torch/_inductor/sizevars.py

```diff
@@ -138,14 +114,14 @@ def visit_modular_indexing(base, divisor, modulus):
                 # actual iteration range is to size-1
                 iter_ranges_zero = {k: 0 for k, v in var_ranges.items()}
                 base_lowest = sympy_subs(base, iter_ranges_zero)
-                if self.maybe_guard_lt(base_lowest, 0):
+                if self.statically_known_lt(base_lowest, 0):
```

Comment:
This looks inverted. ModularIndexing is the safe default, but here you're testing if it's statically known that you can't replace (which seems wrong)

Reply:
This whole function is shady, written at a time when there was no better range analysis, is our range analysis reliable for modulus arithmetic now?

Reply:
I thought this code was around avoiding the sympy infinite loop ModularIndexing?

Reply:
This is attempting to replace (a//b) % c with (a//b) if it's known that (a//b) is always less than c, because this allows for simpler indexing expressions and better generated code

Reply:
Oh I see - so if it is statically know, we don't need to go into MI - because base_lowest being unde r0 means we can just FloorDiv?